### PR TITLE
Added apply-time-mutation to docs

### LINKED ISSUE #42151: Annotation `config.kubernetes.io/apply-time-mutation` not registered / documented
**This is a Bug Report**

**Problem:** https://github.com/kubernetes-sigs/cli-utils/blob/master/README.md mentions an annotation, `config.kubernetes.io/apply-time-mutation`. We haven't documented this in https://kubernetes.io/docs/reference/labels-annotations-taints/ (which means it's not registered).

**Proposed Solution:**
Document that annotation.

**Page to Update:**
https://kubernetes.io/docs/reference/labels-annotations-taints/

**Additional Information:**
/language en
/sig docs

### PR DESCRIPTION
**Pull request by AryanSharma9917**
Description: Documented `config.kubernetes.io/apply-time-mutation` annotation.
This PR fixed #42151

### CODE REVIEW

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 1]
```diff
@@ -262,6 +262,20 @@ This annotation is part of the Kubernetes Resource Model (KRM) Functions Specifi
 which is used by Kustomize and similar third-party tools.
 For example, Kustomize removes objects with this annotation from its final build output.
 
+### config.kubernetes.io/apply-time-mutation
+
+Type: Annotation
+
+Example: `config.kubernetes.io/apply-time-mutation: "true"`
```
Comment (Ritikaa96) [inline content/en/docs/reference/labels-annotations-taints/_index.md:269]: Are you sure it's value should be in boolean? 
Please read : https://github.com/kubernetes-sigs/cli-utils/blob/master/README.md#apply-time-mutation
Let's correct the values first

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 2]
```diff
@@ -262,6 +262,20 @@ This annotation is part of the Kubernetes Resource Model (KRM) Functions Specifi
...
+
+Used on: All objects
+
+The `config.kubernetes.io/apply-time-mutation` annotation is used in manifests to indicate whether an object may undergo mutation during the apply-time process, and thus, should not be directly modified by users or external tooling.
+
+A value of `"true"` for this annotation signifies that the object's configuration might be altered during the application process, either by server-side controllers or by client-side tools like kubectl apply. It serves as a warning to users that any changes made directly to this object may be overridden during the apply process.
+
+On the other hand, a value of `"false"` can be used to indicate that the object's configuration should not be subject to any automatic mutations during apply-time. This means that the object should retain its exact configuration as specified by the user or external tool.
```
Comment (Ritikaa96) [inline content/en/docs/reference/labels-annotations-taints/_index.md:277]:
``` suggestion
This annotation is used on the target object to be modified to configure [Apply-Time Mutation](https://github.com/kubernetes-sigs/cli-utils/blob/master/README.md#apply-time-mutation). The annotation may specify one or more substitutions. 
Each substitution includes a source object, a source field path, and a target field path, with an optional token.
The source and target field paths are specified using JSONPath.
If the token is specified, the token is replaced in the target field value string with the source field value. 
If the token is not specified, the target field value is replaced with the source field value. This alternatively allows either templated interpretation or type preservation.

```

**Review by Ritikaa96 — CHANGES_REQUESTED** [review]

**Comment by Ritikaa96** [comment]
Also have a look at this : https://kpt.dev/reference/annotations/
for other annotations.

**Comment by sftim** [comment]
Thanks @Ritikaa96. As a reviewer if you see a PR where the text looks like it's based on the author (or a tool they use) imagining how things work, it's OK to call that out.

@AryanSharma9917 if you did use a computer to guess at the text to use, please bear in mind that a human still needs to review your work - even if it's plausible-looking but very wrong.

**Comment by AryanSharma9917** [comment]
> Thanks @Ritikaa96. As a reviewer if you see a PR where the text looks like it's based on the author (or a tool they use) imagining how things work, it's OK to call that out.
> 
> @AryanSharma9917 if you did use a computer to guess at the text to use, please bear in mind that a human still needs to review your work - even if it's plausible-looking but very wrong.

Okey @Ritikaa96 @sftim I'll look into it, and thanks for pointing out ky mistakes!

File: content/en/docs/reference/labels-annotations-taints/_index.md (hunk 1 above)
Comment (AryanSharma9917) [reply content/en/docs/reference/labels-annotations-taints/_index.md]: Hi @Ritikaa96, I am getting confused with the example value for config.kubernetes.io/apply-time-mutation annotation is a multiline string in YAML format, specifying a list of substitutions, each represented by a dictionary (or map). Each substitution specifies the sourceRef, sourcePath, targetPath, and optionally the token. So how do I mention that out or do I simply leave blank example value.
```
metadata:
  annotations:
    config.kubernetes.io/apply-time-mutation: |
      - sourceRef:
          kind: Pod
          name: pod-b
        sourcePath: $.status.podIP
        targetPath: $.spec.containers[?(@.name=="nginx")].env[?(@.name=="SERVICE_HOST")].value
        token: ${pod-b-ip}
      - sourceRef:
          kind: Pod
          name: pod-b
        sourcePath: $.spec.containers[?(@.name=="nginx")].ports[?(@.name=="tcp")].containerPort
        targetPath: $.spec.containers[?(@.name=="nginx")].env[?(@.name=="SERVICE_PORT")].value
        token: ${pod-b-port}
```
In this example, the multiline string contains two substitutions, each represented by a dictionary (indicated by the - before each item). Each dictionary specifies the sourceRef, sourcePath, targetPath, and token properties, which are necessary for the Applier to perform the field value substitution during the apply process. The ${pod-b-ip} and ${pod-b-port} are placeholders that will be replaced with actual values based on the source object pod-b during the apply process.

**Comment by AryanSharma9917** [comment]
Hey @sftim, I have made the changes in the PR could you please take a look?

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 3]
```diff
@@ -262,6 +262,20 @@ This annotation is part of the Kubernetes Resource Model (KRM) Functions Specifi
 which is used by Kustomize and similar third-party tools.
 For example, Kustomize removes objects with this annotation from its final build output.
 
+### config.kubernetes.io/apply-time-mutation
+
+Type: Annotation
+
+Example: `config.kubernetes.io/apply-time-mutation: "pods"`
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:269]: Try this:
```suggestion
Example:
```yaml
    config.kubernetes.io/apply-time-mutation: |
      - sourceRef:
          kind: Pod
          name: pod-a
        sourcePath: $.status.podIP
        targetPath: $.spec.containers[?(@.name=="example")].env[?(@.name=="SERVICE_HOST")].value
        token: ${pod-a-ip}
```

I think that's a good start

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 4]
```diff
@@ -262,6 +262,20 @@ This annotation is part of the Kubernetes Resource Model (KRM) Functions Specifi
...
+
+Used on: All objects
+
+This annotation is used on the target object to be modified to configure [Apply-Time Mutation](https://github.com/kubernetes-sigs/cli-utils/blob/master/README.md#apply-time-mutation). The annotation may specify one or more substitutions.
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:273]:
```suggestion
This annotation is used on the target object to be modified to configure [Apply-Time Mutation](https://kpt.dev/reference/annotations/apply-time-mutation/). The annotation may specify one or more substitutions.
```

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 5]
```diff
@@ -262,6 +262,20 @@ This annotation is part of the Kubernetes Resource Model (KRM) Functions Specifi
...
+
+Example: `config.kubernetes.io/apply-time-mutation: "pods"`
+
+Used on: All objects
+
+This annotation is used on the target object to be modified to configure [Apply-Time Mutation](https://github.com/kubernetes-sigs/cli-utils/blob/master/README.md#apply-time-mutation). The annotation may specify one or more substitutions. 
+Each substitution includes a source object, a source field path, and a target field path, with an optional token.
+The source and target field paths are specified using JSONPath.
+If the token is specified, the token is replaced in the target field value string with the source field value. 
+If the token is not specified, the target field value is replaced with the source field value. This alternatively allows either templated interpretation or type preservation.
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:277]: You can leave most of this detail out - people can click on the link to learn more.

**Comment by AryanSharma9917** [comment]
Hey @sftim, could you please review my PR.

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 6]
```diff
@@ -262,6 +262,28 @@ This annotation is part of the Kubernetes Resource Model (KRM) Functions Specifi
...
+          name: example-config
+        sourcePath: $.metadata.annotations.app\.kubernetes\.io/managed-by
+        targetPath: $.spec.containers[?(@.name=="example")].env[?(@.name=="CONFIG_MANAGED_BY")].value
+```
+
+Used on: All objects
+
+This annotation is used on the target object to be modified to configure [Apply-Time Mutation](https://kpt.dev/reference/annotations/apply-time-mutation/). The annotation may specify one or more substitutions. 
+Each substitution includes a source object, a source field path, and a target field path, with an optional token.
+If the token is not specified, the target field value is replaced with the source field value. This alternatively allows either templated interpretation or type preservation.
```
Comment (Ritikaa96) [inline content/en/docs/reference/labels-annotations-taints/_index.md:285]: As stated by @sftim, you can leave out most of the info in favor of the link provided.
```suggestion

```

**Comment by AryanSharma9917** [comment]
Hey @sftim is my PR ready to be merged?

**Comment by AryanSharma9917** [comment]
Hey @bashofmann , @sftim  , @Ritikaa96  is my PR ready to be merged?

**Comment by divya-mohan0209** [comment]
Hey @AryanSharma9917 : This PR has merge conflicts and needs a rebase. Please could you address them?

**Comment by kbhawkey** [comment]
Hello @AryanSharma9917 .
Can you rebase your pull request?
Thanks

**Comment by divya-mohan0209** [comment]
Hello @AryanSharma9917 , thank you for your contribution. Since there has been no response from your side for ~3 months on this PR despite reminders, I'll be closing it off. If (and when) you have the bandwidth, please feel free to resume work by rebasing your branch and addressing comments.

/close

**Pull request closed without merging**